compiler/kernel/bufferfilter.go

```diff
@@ -38,23 +37,27 @@ func CompileBufferFilter(e dag.Expr) (*expr.BufferFilter, error) {
 			return expr.NewAndBufferFilter(left, right), nil
 		}
 		if e.Op == "or" {
-			left, err := CompileBufferFilter(e.LHS)
+			left, err := CompileBufferFilter(zctx, e.LHS)
 			if err != nil {
 				return nil, err
 			}
-			right, err := CompileBufferFilter(e.RHS)
+			right, err := CompileBufferFilter(zctx, e.RHS)
 			if left == nil || right == nil || err != nil {
 				return nil, err
 			}
 			return expr.NewOrBufferFilter(left, right), nil
 		}
 		return nil, nil
 	case *dag.Search:
-		if e.Value.Type == "net" || e.Value.Type == "regexp" {
+		val, err := zson.ParseValue(zctx, e.Value)
+		if err != nil {
+			return nil, err
+		}
+		if val.Type == zed.TypeNet {
 			return nil, nil
 		}
-		if e.Value.Type == "string" {
-			pattern := norm.NFC.Bytes(zed.UnescapeBstring([]byte(e.Value.Text)))
+		if val.Type == zed.TypeString {
```

Comment:
Unlikely as there are to appear here, we should probably armor this against named types.
```go
		switch zed.AliasOf(val.Type) {
		case zed.TypeNet:
			return nil, nil
		case zed.TypeString:
```